### PR TITLE
Fix pdapi cache when cluster TLS enabled

### PR DESCRIPTION
**Pull request by weekface**
### What problem does this PR solve? 

fixes: https://github.com/pingcap/tidb-operator/issues/2253

### Check List 

Tests 

 - Unit test
 - E2E test
 - Stability test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has Go code change

Side effects

Related changes

 - Need to cherry-pick to the release branch

### Does this PR introduce a user-facing change?:

```release-note
Fix pdapi cache issue while cluster TLS is enabled
```

### CODE REVIEW

File: pkg/pdapi/pdapi.go [hunk 1]
```diff
@@ -67,6 +67,7 @@ func NewDefaultPDControl(kubeCli kubernetes.Interface) PDControlInterface {
 // It loads in-cluster root ca if caCert is empty.
 func GetTLSConfig(kubeCli kubernetes.Interface, namespace Namespace, tcName string, caCert []byte) (*tls.Config, error) {
 	secretName := util.ClusterClientTLSSecretName(tcName)
+	// TODO use secretLister instead of kubeCli.CoreV1().Secrets().Get() to optimize the performance
```
Comment (cofyc) [inline pkg/pdapi/pdapi.go:70]: this is intended, see https://github.com/pingcap/tidb-operator/issues/1327

Comment (weekface) [reply pkg/pdapi/pdapi.go]: got it.

File: pkg/controller/tidb_control.go [hunk 2]
```diff
@@ -53,6 +54,10 @@ type TiDBControlInterface interface {
 
 // defaultTiDBControl is default implementation of TiDBControlInterface.
 type defaultTiDBControl struct {
+	// Now, all controllers use this single tidbConrol instance,
+	// we should add a mutex to avoid two controllers update the httpClient.Transport at same time.
+	mutex sync.Mutex
+	// TODO use httpClients cache instead, like pkg/pdapi/pdapi.go does.
```
Comment (cofyc) [inline pkg/controller/tidb_control.go:60]: the cache mechanism in `pdapi` has an issue that stale clients will not be freed when the TC cluster is gone. If we want to optimize this, we can cache PD/TiDB clients (per TiDB cluster CR) in the tidb cluster controller instead and free on the deletion of the tidb cluster.

Comment (weekface) [reply pkg/controller/tidb_control.go:60]: Cache the whole client per TiDB cluster in tidb cluster controller is a good idea.

**Pull request merged**